splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/actions/DDLConstantOperation.java

```diff
@@ -58,549 +61,568 @@
-	 *  statement. Because of these differences between constraints and views
-	 *  (and triggers), there are 2 different methods in this class to save
-	 *  their privileges in the dependency system.
-	 *
-	 *  For each required privilege, we now register of a dependency on a role
-	 *  if that role was required to find an applicable privilege.
-	 *
-	 *  @param activation The execution environment for this constant action.
-	 *  @param dependent Make this object depend on required privileges
-	 *
-	 * @exception StandardException		Thrown on failure
-	 */
-	protected void storeViewTriggerDependenciesOnPrivileges( Activation activation, Dependent dependent)
-			throws StandardException {
-		SpliceLogUtils.trace(LOG, "storeViewTriggerDependenciesOnPrivileges with dependent %s",dependent);
-		LanguageConnectionContext lcc = activation.getLanguageConnectionContext();
-		DataDictionary dd = lcc.getDataDictionary();
-		DependencyManager dm = dd.getDependencyManager();
-		String dbo = dd.getAuthorizationDatabaseOwner();
+                                // Role has been relied upon, so register a
+                                // dependency.
+                                permDesc = findRoleUsage(activation, statPerm);
+                                if (permDesc != null) {
+                                    dm.addDependency(dependent, permDesc,lcc.getContextManager());
+                                    trackRoleDependency(activation, dependent, permDesc, dependedRoleList);
+                                }
+                            }
+                        }
+                    }
+
+                    if (!(statPerm instanceof StatementRoutinePermission)) {
+                        //We have found the REFERENCES privilege for all the
+                        //columns in foreign key constraint and we don't
+                        //need to go through the rest of the privileges
+                        //for this sql statement.
+                        break;
+                    } else {
+                        // For EXECUTE privilege there may be several functions
+                        // referenced in the constraint, so continue looking.
+                    }
+                }
+            }
+        }
+
+    }
+
+    /**
+     * Notify other region servers of potential metadata changes
+     * @param tc
+     */
+    public void notifyMetadataChanges(TransactionController tc, List<DDLMessage.DDLChange> changes) throws StandardException {
+        if (changes.size() == 1) {
+            notifyMetadataChanges(tc, changes.get(0));
+        } else if (changes.size() > 1) {
+            notifyMetadataChanges(tc, ProtoUtil.createMultiChange(changes.get(0).getTxnId(), changes));
+        }
+    }
+
+    /**
+     * Notify other region servers of potential metadata changes
+     * @param tc
+     */
+    public void notifyMetadataChanges(TransactionController tc, DDLMessage.DDLChange change) throws StandardException {
```

Comment:
Maybe use singular form in function name like `notifyMetadataChange` since this function deals with one change only. But it's a minor thing.